### PR TITLE
Proper HashMaps for Skylerberg, extend the mpsc api with a try_recv() method, add tests for mpsc, fix possible (but unlikely) premature drop in mpsc.

### PR DESCRIPTION
**Pull request by Ticki**
cc @skylerberg

### CODE REVIEW

**Comment by Ticki** [comment]
ping @jackpot51

**Comment by jackpot51** [comment]
Conflicts with libredox rename, can you fix those?

**Comment by Ticki** [comment]
@jackpot51 Done.

File: filesystem/schemes/orbital/session.rs [hunk 1]
```diff
@@ -174,6 +178,19 @@ impl Session {
...
+
+                if self.shutdown.has_data() {
+                    x = self.display.width as i32 - self.shutdown.width() as i32;
+                    let y = self.display.height as isize - self.shutdown.height() as isize;
+                    if mouse_event.y >= y as i32 && mouse_event.x >= x &&
+                       mouse_event.x < x + self.shutdown.width() as i32 {
+                        File::create("acpi:off");
+                    }
+                }
+>>>>>>> 0722138ddfb10642d6d33d15a2456bca511dfe15
```
Comment (jackpot51) [inline filesystem/schemes/orbital/session.rs]: Merge not completed here

Comment (Ticki) [inline filesystem/schemes/orbital/session.rs]: done.

**Comment by jackpot51** [comment]
good job

**Pull request merged**